src/main/java/racingcar/service/Race.java

```diff
@@ -1,64 +1,38 @@
 package racingcar.service;
 
-import static racingcar.config.RacingCarException.PLAYING_COUNT_OUT_OF_RANGE;
-
 import java.util.List;
 import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 
-import racingcar.domain.CarNames;
-import racingcar.domain.Cars;
-import racingcar.domain.MovementStrategy;
+import racingcar.domain.Rounds;
+import racingcar.domain.cars.Cars;
+import racingcar.domain.movement.MovementStrategy;
 import racingcar.vo.CarResult;
 import racingcar.vo.GameResult;
 import racingcar.vo.RoundResult;
 
 public class Race {
 
-    private static final int MINIMUM_PLAYING_COUNT = 1;
-    private static final int START_ROUND = 1;
-
-    private final Cars cars;
-    private final MovementStrategy movementStrategy;
-
-    private Race(final Cars racingCars, final MovementStrategy movementStrategy) {
-        this.cars = racingCars;
-        this.movementStrategy = movementStrategy;
-    }
-
-    public GameResult progress(final int playingCount) {
-        validatePlayingCountIsInRange(playingCount);
-
-        return new GameResult(runAllRounds(playingCount), this.cars.winnerNames());
-    }
+    public GameResult progress(final Cars cars, final Rounds rounds, final MovementStrategy movementStrategy) {
+        final List<RoundResult> roundResults = runAllRounds(cars, rounds, movementStrategy);
+        final Cars winners = cars.winners();
 
-    private static void validatePlayingCountIsInRange(final int playingCount) {
-        if (isPlayingCountOutOfRange(playingCount)) {
-            throw new IllegalArgumentException(PLAYING_COUNT_OUT_OF_RANGE.message(playingCount));
-        }
+        return new GameResult(roundResults, CarResult.fromCars(winners));
     }
 
-    private static boolean isPlayingCountOutOfRange(int playingCount) {
-        return playingCount < MINIMUM_PLAYING_COUNT;
+    private List<RoundResult> runAllRounds(
+            final Cars cars,
+            final Rounds rounds,
+            final MovementStrategy movementStrategy
+    ) {
+        return rounds.rounds()
+                .stream()
+                .map(round -> runSingleRound(round, cars, movementStrategy))
+                .collect(Collectors.toUnmodifiableList());
```

Comment:
```suggestion
        return rounds.map(round -> runSingleRound(round, cars, moveStrategy));
```

이런식으로 고차함수를 전달하는 방식으로도 구현가능할 것 같네요